src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -295,6 +295,7 @@ class ShenandoahHeap : public CollectedHeap, public ShenandoahSpaceInfo {
 private:
   bool _gc_state_changed;
   ShenandoahSharedBitmap _gc_state;
+  ShenandoahSharedFlag   _heap_changed;
```

Comment:
The documentation at `has_changed()` is good, but may be a single small documentation comment here, perhaps something like:

```
   ShenandoahSharedFlag _heap_changed;  // set if heap has changed since last check
```